find-phantom-resource.ps1

```
@@ -351,13 +386,206 @@ function Test-ResourceOrphaned {
+| project name, impact=properties.impact, description=properties.shortDescription.solution, resourceId=properties.resourceMetadata.resourceId
+"@
+    }
+
+    $totalOrphans = 0
+    foreach ($queryName in $queries.Keys) {
+        Write-Host "`n=== $queryName ===" -ForegroundColor Cyan
+        $kql = $queries[$queryName]
+        try {
+            $resultJson = az graph query -q $kql --first 200 -o json --only-show-errors 2>&1
+            if ($LASTEXITCODE -ne 0) {
+                Write-Host "  Query failed: $resultJson" -ForegroundColor Red
```

PSScriptAnalyzer (1.24.0)

[warning] File 'find-phantom-resource.ps1' uses Write-Host. Avoid using Write-Host because it might not work in all hosts, does not work when there is no host, and (prior to PS 5.0) cannot be suppressed, captured, or redirected. Instead, use Write-Output, Write-Verbose, or Write-Information.

(PSAvoidUsingWriteHost)